docs/topics/data-analysis/data-analysis-work-with-data-sources.md

```diff
@@ -1,22 +1,179 @@
 [//]: # (title: Retrieve data from files)
 
-Simple guide how to start working data sources with Kotlin notebooks.
+Kotlin Notebook, coupled with the  [DataFrame library](https://kotlin.github.io/dataframe/gettingstarted.html), enables 
+you to work with both non-structured and structured data. It offers the flexibility to transform non-structured data, 
+such as those found in TXT files, into structured datasets. For this transformation, you can utilize methods 
+such as [`add`](https://kotlin.github.io/dataframe/adddf.html), [`split`](https://kotlin.github.io/dataframe/split.html),
+[`convert`](https://kotlin.github.io/dataframe/convert.html), and [`parse`](https://kotlin.github.io/dataframe/parse.html). 
+Additionally, this toolset enables the retrieval and manipulation of data from various structured file formats, 
+including CSV, JSON, XLS, XLSX, and Apache Arrow.
 
 ## Before you start
 
-* Download and install the latest version of [IntelliJ IDEA](https://www.jetbrains.com/idea/download/index.html).
-* Install the [Kotlin Notebook plugin](https://plugins.jetbrains.com/plugin/16340-kotlin-notebook).
-* Add a dependency to the DataFrame library (?)
+1. Download and install the latest version of [IntelliJ IDEA Ultimate](https://www.jetbrains.com/idea/download/?section=mac).
+2. Install the [Kotlin Notebook plugin](https://plugins.jetbrains.com/plugin/16340-kotlin-notebook) in IntelliJ IDEA.
 
-## From CSV file
+   > Alternatively, access the Kotlin Notebook plugin from **Settings/Preferences** | **Plugins** | **Marketplace** within IntelliJ IDEA.
+   >
+   {type="tip"}
 
-## From JSON file
+3. Create a new Kotlin Notebook by selecting **File** | **New** | **Kotlin Notebook**.
+4. In the Kotlin Notebook, import the Kotlin DataFrame library by running the following command:
 
-## From XML file
+```kotlin
+%use dataframe
+```
 
-## From TXT file
+## Retrieve data from a file
+
+To retrieve data from a file in Kotlin Notebook:
+
+1. **Open Kotlin Notebook file:** Begin by opening your Kotlin Notebook file (`.ipynb`).
+2. **Import libraries:** Import the DataFrame library by adding `%use dataframe` in a code cell at the start of your Notebook.
+> Make sure to run the code cell with the `%use dataframe` line before you run any other code cells that rely on the DataFrame library.
+>
+{type="note"}
+
+3. **Read data from a file:** Use the `.read()` function from DataFrame to retrieve data. For example, to read a CSV file, 
+use: `DataFrame.read("example.csv")`.
+
+The `.read()` function automatically detects the input format based on the file extension and content.
+You can also add other arguments to customize the function, such as specifying the delimiter with `delimiter = ';'`.
+
+For a comprehensive overview of additional file formats and a variety of read functions, see the 
+[DataFrame library documentation](https://kotlin.github.io/dataframe/read.html).
+
+## Display data
```

Comment:
Hmm 🤔, I think since this is a general guide, "Display data" works better for a broader, non-specific data set (also it aligns better with the other titles in my opinion)